src/components/heatmap/heatmap.tsx

```diff
@@ -0,0 +1,344 @@
+   * title: title of heatmap
+   * }
+   * ----------
+   * Layout: 
+   * height
+   * margin: {t, r, b, }
+   * title
+   * width
+   * xaxis: {tickangle, tickfont, tickvals, ticktext}
+   * yaxis: {tickangle, tickvals, ticktext, tickfont} 
+   */
+
+  const createPlotLayoutParams = (plot: any = {}) => {
+    const configPresentation = Object.assign({}, config.presentation);
+    let layout = {};
+    if (plot) {
+      const longestXTick = plot?.rows?.x.reduce((a: any, b: any) => { return a?.length > b?.length ? a : b; });
+      const longestYTick = plot?.rows?.y.reduce((a: any, b: any) => { return a?.length > b?.length ? a : b; });
+      const inputParams = {
+        width: typeof configPresentation.width !== 'undefined' ? configPresentation.width : 1200,
+        xTickAngle: typeof configPresentation.xTickAngle !== 'undefined' ? configPresentation.xTickAngle : 50,
+        tickFont: {
+          family: typeof configPresentation.tickFontFamily !== 'undefined' ? configPresentation.tickFontFamily : 'Courier',
+          size: typeof configPresentation.tickFontSize !== 'undefined' ? configPresentation.tickFontSize : 12
+        }
+      };
+      const layoutParams = getHeatmapLayoutParams(inputParams, longestXTick?.length, longestYTick?.length, plot.rows?.y?.length);
+
+      layout = {
+        title: plot.title,
+        xaxis: {
+          tickangle: layoutParams.xTickAngle,
+          tickfont: layoutParams.tickFont,
+          tickvals: plot.rows?.x,
+          ticktext: plot.rows?.x,
+        },
+        yaxis: {
+          tickangle: layoutParams.yTickAngle,
+          tickvals: plot.rows?.y,
+          ticktext: plot.rows?.y,
+          tickfont: layoutParams.tickFont
+        },
+        margin: layoutParams.margin,
+        height: layoutParams.height,
+        width: layoutParams.width
+      };
+      if (plot.rows) {
+        plot['rows']['colorbar'] = {
+          lenmode: 'pixels',
+          len: layoutParams.height - 40 < 100 ? layoutParams.height - 40 : 100
+        }
+      }
+    }
+    return { plot, layout };
+  }
+
+
+  return (
+    <div style={{ marginTop: '1%', overflow: 'auto', maxHeight: '90vh' }}>
+      <ChaiseToolTip
+        placement='right'
+        tooltip='View Array Data related to this Gene'
+      >
+        <a style={{ marginLeft: '1%' }}
+          href={`${window.origin}/chaise/recordset/${ConfigService.ERMrest.createPath('2', 'Gene_Expression', 'Array_Data', facet)}
+          ?pcid=${Object(header).cid}&ppid=${Object(header).pid}`} target='_blank' rel='noreferrer'>
+          View Array Data Table</a>
+      </ChaiseToolTip>
+      {/* Iterating over the heatmaps fetched for the given NCBIGeneID */}
+      {heatmaps.map((heatmap: any): JSX.Element => {
+        const { plot, layout } = createPlotLayoutParams(heatmap);
+        return <div id={`heatmap_${heatmap.id}`} key={`heatmap_${heatmap.id}`}>
+          <HeatmapPlot key={`heatmap_${heatmap.id}`} data={plot} layout={layout} />
```

Comment:
Does this need a key too since the parent `<div>` that is being repeated has the same `key` value defined?